src/bcm-nexus/view-backend.cpp

```diff
@@ -65,21 +103,32 @@ struct ViewBackend : public IPC::Host::Handler, public WPE::LibinputServer::Clie
 
     void commitBuffer(uint32_t, uint32_t);
 
+#ifdef KEY_INPUT_HANDLING_LIBINPUT
     // WPE::LibinputServer::Client
     void handleKeyboardEvent(struct wpe_input_keyboard_event*) override;
     void handlePointerEvent(struct wpe_input_pointer_event*) override;
     void handleAxisEvent(struct wpe_input_axis_event*) override;
     void handleTouchEvent(struct wpe_input_touch_event*) override;
+#endif
 
     struct wpe_view_backend* backend;
     IPC::Host ipcHost;
 
+#ifdef KEY_INPUT_HANDLING_WAYLAND
+    Wayland::Display& m_display;
+    struct wl_surface* m_surface { nullptr };
+    struct wl_shell_surface *m_shellSurface { nullptr };
```

Comment:
m_surface and m_shellSurface should be cleaned up in the deconstructor (though I don't know if m_shellSurface is a standalone resource).